src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToFhirService.java

```diff
@@ -1,14 +1,53 @@
 package uk.nhs.digital.nhsconnect.nhais.service;
 
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Interchange;
+
+import org.hl7.fhir.r4.model.Organization;
 import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
 import org.springframework.stereotype.Component;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.Interchange;
 
 @Component
 public class EdifactToFhirService {
 
-    Parameters convertToFhir(Interchange interchange) {
-        return new Parameters();
+    public Parameters convertToFhir(Interchange interchange) {
+        Patient patient = createPatient("example");
+
+        Reference managingOrganization = createManagingOrganizationReference(interchange);
+        Reference generalPractitioner = createGeneralPractitionerReference(interchange);
+
+        patient.setManagingOrganization(managingOrganization);
+        patient.addGeneralPractitioner(generalPractitioner);
+
+        return createParameters(patient);
     }
 
+    private Parameters createParameters(Resource... resources) {
+        Parameters parameters = new Parameters();
+        for (Resource resource : resources) {
+            parameters.addParameter(new Parameters.ParametersParameterComponent().setResource(resource));
```

Comment:
Shouldn't parameters be named? I think the example had a name on it

Reply:
They can be, but naming works only on value params, so it's useless for resources

Reply:
Added for clarity